espnet2/utils/plot_sinc_filters.py

```diff
@@ -0,0 +1,260 @@
+            img_path = os.path.join(args.out_folder, img_name)
+            plt.savefig(img_path, bbox_inches="tight")
+            print("Plotted %s" % img_path)
+
+            kernel = kernels[i][0]
+            plt.clf()
+            plt.xticks([])
+            plt.yticks([])
+            plt.plot(x, kernel)
+            img_name = "filter_kernel_%s.%s" % (str(i).zfill(2), args.filetype)
+            img_path = os.path.join(args.out_folder, img_name)
+            plt.savefig(img_path, bbox_inches="tight")
+            print("Plotted %s" % img_path)
+
+            plt.clf()
+            plt.xlabel("kernel index")
+            plt.plot(x, kernel)
+            plt.plot(x, pre_kernel, "--", alpha=0.5)
+            img_name = "filter_kernel_both_%s.%s" % (str(i).zfill(2), args.filetype)
+            img_path = os.path.join(args.out_folder, img_name)
+            plt.savefig(img_path, bbox_inches="tight")
+            print("Plotted %s" % img_path)
+
+            y = np.zeros_like(x_f)
+            y[F_mins[i] : F_maxs[i]] = 1.0
+            plt.clf()
+            plt.plot(x_f, y)
+            img_name = "filter_freq_%s.%s" % (str(i).zfill(2), args.filetype)
+            img_path = os.path.join(args.out_folder, img_name)
+            plt.savefig(img_path, bbox_inches="tight")
+            print("Plotted %s" % img_path)
+
+            pre_y = np.zeros_like(x_f)
+            pre_y[pre_F_mins[i] : pre_F_maxs[i]] = 1.0
+            plt.clf()
+            plt.plot(x_f, y)
+            plt.plot(x_f, pre_y)
+            img_name = "filter_freq_both_%s.%s" % (str(i).zfill(2), args.filetype)
+            img_path = os.path.join(args.out_folder, img_name)
+            plt.savefig(img_path, bbox_inches="tight")
+            print("Plotted %s" % img_path)
+
+    plt.clf()
+    filters = [32, 71, 113, 126]
+    fig, axs = plt.subplots(2, 2, sharex=True, sharey="row")
+
+    axs[0, 0].plot(x, kernels[filters[0]][0])
+    axs[0, 0].plot(x, pre_kernels[filters[0]][0], "--", alpha=0.5)
+    axs[0, 1].plot(x, kernels[filters[1]][0])
+    axs[0, 1].plot(x, pre_kernels[filters[1]][0], "--", alpha=0.5)
+    axs[1, 0].plot(x, kernels[filters[2]][0])
+    axs[1, 0].plot(x, pre_kernels[filters[2]][0], "--", alpha=0.5)
+    axs[1, 1].plot(x, kernels[filters[3]][0])
+    axs[1, 1].plot(x, pre_kernels[filters[3]][0], "--", alpha=0.5)
+
+    img_name = "filter_kernel_ensemble2.%s" % (args.filetype)
+    img_path = os.path.join(args.out_folder, img_name)
+    plt.savefig(img_path, bbox_inches="tight")
+    print("Plotted %s" % img_path)
+
+
+def main(argv):
+    """Load the model, generate kernel and bandpass plots."""
+    parser = get_parser()
+    args = parser.parse_args(argv)
+    model_path = args.model_path
+    sample_rate = args.sample_rate
+
+    model = torch.load(model_path, map_location="cpu")
+    if "model" in model:  # snapshots vs. model.acc.best
+        model = model["model"]
+    filters = model["preencoder.filters.f"].detach().cpu().numpy()
```

Comment:
Add type check for preencoder.

Reply:
Added directly a type check for the filter values for more flexibility.

Reply:
I can't allow such flexibility. It's d dangerous design.
Please add type check. We must add type check if new class is added.

Reply:
Oh, sorry, you don't instantiated the model class, so we can't check the type.
Could you add checking the existence of the attribute?

Reply:
Added the check for the key in the model dictionary.